docs/dev/generator.md

```diff
@@ -0,0 +1,201 @@
+# Importer Documentation
+
+This document details the importer process.
+
+## Overview
+
+The implementation is divided into two components: generic and NetBox specific.
+
+### Generic Component
+
+The generic component encompasses the `NautobotAdapter` and `SourceAdapter` capable of generating `DiffSyncModel` classes based on the input data, the current version of Nautobot, and any deviations defined in the source configuration.
+
+It comprises the following modules, located in the `nautobot_netbox_importer/generator` directory:
+
+- `base.py`: Implements base classes and utilities utilized throughout the application, also establishing constants and shared type definitions.
+- `nautobot.py`: Provides the `NautobotAdapter`, along with `NautobotModelWrapper` and `NautobotFieldWrapper` classes that define the Nautobot structure.
+- `source.py`: Offers the `SourceAdapater`, `SourceModelWrapper` and `SourceField` classes that define the source structure.
+- `fields.py`: Contains factories for fields deviation definitions.
+- `summary.py`: Supplies utility functions for summarizing data structures and presenting import statistics.
+
+### NetBox Specific Component
+
+The NetBox specific component is further segmented into sections, located in the `nautobot_netbox_importer/diffsync` directory:
+
+- `adapters/nautobot.py`: Inherits `NautobotAdapter` from the generic `NautobotAdapter` and implements pieces necessary for SSoT job.
+- `adapters/netbox.py`: Inherits `NetBoxAdapter` from the generic `SourceAdapter` and implements the data reader and importer that facilitate the transition from NetBox to Nautobot.
+- `models/` directory: Contains individual module files that define deviations and field mappings from NetBox to Nautobot.
+
+## Stages
+
+The import process consists of the following stages:
+
+### Defining a Source Data
+
+The initial step involves creating a `SourceAdapter()`. It accepts an argument, `get_source_data`, which is `Callable` that returns `Iterable` of the source data items. Each source data item is encapsulated in `SourceRecord(content_type: ContentTypeStr, data: Dict)` instances. `SourceAdapter` constructor also passes any additional arguments to its ancestor, the `DiffSync` class.
+
+The data undergoes two cycles: the first to establish the structure and the second to import the actual data.
```

Comment:
could you provide a structure with these two cycles and the sub steps involved?

Reply:
Steps are described in this document, updated this doc to mention that.